app/models/gids_redis.rb

```diff
@@ -36,7 +44,17 @@ def response_from_redis_or_service
     end
   end
 
+  def response_from_redis_or_search_service
+    do_cached_with(key: rest_call.to_s + scrubbed_params.to_s) do
+      gi_search_service.send(rest_call, scrubbed_params)
```

Comment:
Wondering if this can be refactored/cleaned up somehow. This code is almost identical to response_from_redis_or_service minus the use of gi_search_service vs  gi_service

Reply:
refactored